File: 100-Days-Series/Day 91/CatalanNumbers.java
class CatalanNumbers{
    // Method to calculate factorial
    public static int getFactorial(int n){
        if(n==0 || n==1)
            return 1;
        return n*getFactorial(n-1);
    }
    // Method to calculate the nth Catalan number
    public static int getCatalan(int n){
        return getFactorial(2*n)/(getFactorial(n+1)*getFactorial(n));
    }
    // Main Method
    public static void main(String [] args){
        // Let's Start Here
        int n;
        java.util.Scanner sc=new java.util.Scanner(System.in);
        System.out.print("Enter the number of terms (N) : ");
        n=sc.nextInt();
        System.out.println("Catalan Numbers up to "+n+" terms : ");
        for(int i=0;i<n;i++){
            System.out.print(getCatalan(i)+" ");
        }
    }
}

File: 100-Days-Series/Day 91/CatalanNumbers.cpp
#include<iostream>
using namespace std;
 // Function to calculate factorial
int factorial(int n){
    if(n==1 || n==0)
        return 1;
    return n*factorial(n-1);
}
// Function to calculate the nth Catalan number
int getCatalan(int n){
    return factorial(n*2)/(factorial(n+1)*factorial(n));
}
// Let's Start Here
int main(){
    int n;
    cout<<"\nEnter the number of term (N) : ";
    cin>>n;
    cout<<"\nCatalan Numbers up to "<<n<<" terms : "<<endl;
    for(int i=0;i<n;i++){
        cout<<getCatalan(i)<<" ";
    }
    return 0;
}